structured.js

```diff
@@ -54,6 +54,103 @@
             fn: callback
         };
     }
+    
+    /*
+     * return true if n2 < n1 (according to relatively arbitrary criteria)
+     */
+    function shouldSwap(n1, n2) {
+	if (n1.type < n2.type) {
+	    return false;
+	} else if (n1.type > n2.type) {
+	    return true;
```

Comment:
Can you add some comments as to what is going on in these previous lines? It's not completely clear to me what's being compared and what the expected result is.